neo/SmartContract/GasControl.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Neo.VM;
+
+namespace Neo.SmartContract
+{
+    public partial class GasControl : IExecutionControl
+    {
+        public const long GasFree = 0;
+
+        private readonly long gas_amount;
+        private readonly bool testMode;
+
+        public long GasConsumed { get; private set; } = 0;
+
+        public GasControl(long gas = 0, bool testMode = false)
```

Comment:
Perfect, perfect, @shargon.

The refactor is precise!
It is much more readable.

I agree with it.

The other discussions out of the refactor and the class could be moved to other PR.

Maybe we can rename `GasControl` -> `GasManager` or `GasTuringCompleteManager` or `GasTuringIncompleteManager`...aehuaheuaheua
Turing is just for fun. Maybe `GasManager` or `GasLimitsManager`